tracer/build/_build/CodeGenerators/CallTargetsGenerator.cs

```diff
@@ -0,0 +1,743 @@
+
+                return sig;
+            }
+
+            static string GetSignatureName(int index)
+            {
+                return $"g_callTargets_Sig_{index:000}";
+            }
+
+            static string GetSignatureField(string signatureName, string signature)
+            {
+                return $"WCHAR* {signatureName}[]={{{signature}}};";
+            }
+
+            static string GetCallTarget(CallTargetDefinitionSource definition, string signature, TargetFrameworks tfms, int index)
+            {
+                var min = definition.MinimumVersion;
+                var max = definition.MaximumVersion;
+
+                var typ = $"(WCHAR*)WStr(\"{definition.AssemblyName}\"),(WCHAR*)WStr(\"{definition.TargetTypeName}\"),(WCHAR*)WStr(\"{definition.TargetMethodName}\"),";
+                var sig = $"{signature},{definition.TargetParameterTypes.Count + 1},";
+                var ver = $"{min.Major},{min.Minor},{min.Patch},{max.Major},{max.Minor},{max.Patch},";
+                var asy = $"assemblyName,(WCHAR*)WStr(\"{definition.InstrumentationTypeName}\"),{GetCallTargetKind(definition.IntegrationKind)},{(int)definition.InstrumentationCategory}";
+                var tfm = $",{(uint)tfms}";
+                return $"{{{typ}{sig}{ver}{asy}{tfm}}},";
+            }
+
+            static string GetCallTargetKind(int kind)
+            {
+                /*
+                 enum class CallTargetKind : UINT8
+                {
+                    Default = 0,
+                    Derived = 1,
+                    Interface = 2
+                };
+                 */
+                return kind switch
+                {
+                    0 => "CallTargetKind::Default",
+                    1 => "CallTargetKind::Derived",
+                    2 => "CallTargetKind::Interface",
+                    _ => "ERROR"
+                };
+            }
+        }
+
+        internal static void GenerateJsonFile(Dictionary<CallTargetDefinitionSource, TargetFrameworks> definitions, AbsolutePath outputPath)
+        {
+            var orderedDefinitions = definitions
+                                        .Keys
+                                        .OrderBy(static x => x.IntegrationName)
+                                        .ThenBy(static x => x.AssemblyName)
+                                        .ThenBy(static x => x.TargetTypeName)
+                                        .ThenBy(static x => x.TargetMethodName)
+                                        .ToArray();
+
+            var options = new JsonSerializerOptions();
+            options.IncludeFields = true;
+            options.WriteIndented = true;
+            string jsonString = System.Text.Json.JsonSerializer.Serialize(orderedDefinitions, options);
+
+            if (!Directory.Exists(outputPath)) { Directory.CreateDirectory(outputPath); }
+            var fileName = outputPath / FileNames.DefinitionsJson;
+            File.WriteAllText(fileName, jsonString);
```

Comment:
I think you can serialize this directly, right?
```suggestion
            var fileName = outputPath / FileNames.DefinitionsJson;
            using var fs = File.Create(fileName );
            System.Text.Json.JsonSerializer.Serialize(fs, orderedDefinitions, options);
```